packages/aws-cdk/lib/cli.ts

```diff
@@ -165,134 +168,133 @@ export async function exec(args: string[], synthesizer?: Synthesizer): Promise<n
-      allTopLevel: args.all,
-      patterns: args.STACKS,
-    };
+    const cloudFormation = new Deployments({ sdkProvider, toolkitStackName });
 
     const cli = new CdkToolkit({
       cloudExecutable,
       deployments: cloudFormation,
-      verbose: argv.trace || argv.verbose > 0,
-      ignoreErrors: argv['ignore-errors'],
-      strict: argv.strict,
+      verbose: globalOptions.trace || (globalOptions.verbose ?? 0) > 0,
+      ignoreErrors: globalOptions.ignoreErrors,
+      strict: globalOptions.strict,
       configuration,
       sdkProvider,
     });
 
     switch (command) {
       case 'context':
+        const contextOptions = settings.context ?? {};
         return context({
           context: configuration.context,
-          clear: argv.clear,
-          json: argv.json,
-          force: argv.force,
-          reset: argv.reset,
+          clear: contextOptions.clear,
+          json: globalOptions.json,
+          force: contextOptions.force,
+          reset: contextOptions.reset,
         });
 
       case 'docs':
-        return docs({ browser: configuration.settings.get(['browser']) });
+        return docs({ browser: settings.docs?.browser ?? '' });
 
       case 'doctor':
         return doctor();
 
       case 'ls':
       case 'list':
-        return cli.list(args.STACKS, {
-          long: args.long,
-          json: argv.json,
-          showDeps: args.showDependencies,
+        const listOptions = settings.list ?? {};
+        return cli.list(listOptions.STACKS ?? [], {
+          long: listOptions.long,
+          json: globalOptions.json,
+          showDeps: listOptions.showDependencies,
         });
 
       case 'diff':
         const enableDiffNoFail = isFeatureEnabled(configuration, cxapi.ENABLE_DIFF_NO_FAIL_CONTEXT);
+        const diffOptions = settings.diff ?? {};
         return cli.diff({
-          stackNames: args.STACKS,
-          exclusively: args.exclusively,
-          templatePath: args.template,
-          strict: args.strict,
-          contextLines: args.contextLines,
-          securityOnly: args.securityOnly,
-          fail: args.fail != null ? args.fail : !enableDiffNoFail,
-          stream: args.ci ? process.stdout : undefined,
-          compareAgainstProcessedTemplate: args.processed,
-          quiet: args.quiet,
-          changeSet: args['change-set'],
+          stackNames: diffOptions.STACKS ?? [],
+          exclusively: diffOptions.exclusively,
+          templatePath: diffOptions.template,
+          strict: diffOptions.strict,
+          contextLines: diffOptions.contextLines,
+          securityOnly: diffOptions.securityOnly,
+          fail: diffOptions.fail != null ? diffOptions.fail : !enableDiffNoFail,
+          stream: globalOptions.ci ? process.stdout : undefined,
+          compareAgainstProcessedTemplate: diffOptions.processed,
+          quiet: diffOptions.quiet,
+          changeSet: diffOptions.changeSet,
           toolkitStackName: toolkitStackName,
```

Comment:
This is good already, but ideally the next step would be a strongly typed helper function/method so we can do:

```ts
return cli.diff(diffSettingsToDiffOptions(settings.diff));
```